tests/unit/test_adapter.py

```diff
@@ -546,6 +546,109 @@ def test_list_relations_without_caching_with_non_glue_data_catalog(self, parent_
     def test_parse_s3_path(self, s3_path, expected):
         assert self.adapter._parse_s3_path(s3_path) == expected
 
+    @mock_athena
+    @mock_glue
+    @mock_s3
+    def test_swap_table_with_partitions(self, aws_credentials):
+        self.mock_aws_service.create_data_catalog()
+        self.mock_aws_service.create_database()
+        self.adapter.acquire_connection("dummy")
+        target_table = "target_table"
+        source_table = "source_table"
+        self.mock_aws_service.create_table(source_table)
+        self.mock_aws_service.add_partitions_to_table(DATABASE_NAME, source_table)
+        self.mock_aws_service.create_table(target_table)
+        self.mock_aws_service.add_partitions_to_table(DATABASE_NAME, source_table)
+        self.adapter.swap_table(DATABASE_NAME, source_table, DATABASE_NAME, target_table)
+        assert self.adapter.get_table_location(DATABASE_NAME, target_table) == f"s3://{BUCKET}/tables/{source_table}"
+
+    @mock_athena
+    @mock_glue
+    @mock_s3
+    def test_swap_table_without_partitions(self, aws_credentials):
+        self.mock_aws_service.create_data_catalog()
+        self.mock_aws_service.create_database()
+        self.adapter.acquire_connection("dummy")
+        target_table = "target_table"
+        source_table = "source_table"
+        self.mock_aws_service.create_table_without_partitions(source_table)
+        self.mock_aws_service.create_table_without_partitions(target_table)
+        self.adapter.swap_table(DATABASE_NAME, source_table, DATABASE_NAME, target_table)
+        assert self.adapter.get_table_location(DATABASE_NAME, target_table) == f"s3://{BUCKET}/tables/{source_table}"
+
+    @mock_athena
+    @mock_glue
+    @mock_s3
+    def test_swap_table_with_partitions_to_one_without(self, aws_credentials):
+        self.mock_aws_service.create_data_catalog()
+        self.mock_aws_service.create_database()
+        self.adapter.acquire_connection("dummy")
+        target_table = "target_table"
+        source_table = "source_table"
+        # source table does not have partitions
+        self.mock_aws_service.create_table_without_partitions(source_table)
+
+        # the target table has partitions
+        self.mock_aws_service.create_table(target_table)
+        self.mock_aws_service.add_partitions_to_table(DATABASE_NAME, target_table)
+
+        self.adapter.swap_table(DATABASE_NAME, source_table, DATABASE_NAME, target_table)
+        glue_client = boto3.client("glue", region_name=AWS_REGION)
+
+        target_table_partitions = glue_client.get_partitions(DatabaseName=DATABASE_NAME, TableName=target_table).get(
+            "Partitions"
+        )
+
+        assert self.adapter.get_table_location(DATABASE_NAME, target_table) == f"s3://{BUCKET}/tables/{source_table}"
+        assert len(target_table_partitions) == 0
+
+    @mock_athena
+    @mock_glue
+    @mock_s3
+    def test_swap_table_with_no_partitions_to_one_without(self, aws_credentials):
```

Comment:
Is it `test_swap_table_with_no_partitions_to_one_with` ?

Reply:
yes it is, good catch!